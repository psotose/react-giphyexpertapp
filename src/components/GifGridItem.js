import React from 'react'
import PropTypes from 'prop-types';

const GifGridItem = ( {title, url} ) => {  // se pone props en vez de img xq es lo que es en relidad ( props )
                                          // pero si desestructuro es mas facil de leer
  
  // console.log({id, title, url});
  return (
    <div className="card animate__animated animate__bounce">
      <img src={url} alt={title} />
      <p> { title } </p>  
      {/* { img.title } */}
    </div>
  )
}

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}
export default GifGridItem;
