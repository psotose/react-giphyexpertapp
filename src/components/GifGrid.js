import React from 'react' //, { useState, useEffect } le saco esto xq voy a trabajar en el custom hook
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
// import { getGifs } from '../helpers/getGifs';


export const GifGrid = ({ category }) => {
  
  // const [count, setCount] = useState(0);
  // const state = useFetchGifs();
  const { data:images, loading} = useFetchGifs( category );
  // con data:images como quien dice renombro a data y luego puedo llamarlo como images
  //const [images, setImages] = useState([]);

  // useEffect( () => {
  //   getGifs( category )
  //     .then( imgs => setImages( imgs )); // esto es lo mismo que .then(setImages)
  // }, [ category ]) // con el [] le estoy diciendo que solo quiero que se ejecute la instrucci√≥n cuando el componente es renderizado por 1ra vez

  return (
    <>
      <h3 className="animate__bounceInRight"> {category} </h3>
      {/* { loading ? 'Cargando...' : 'Fin de carga' } */}
      { loading && <p className="animate_animated animate_flash">'Cargando...'</p> }  
      {/* con el && solo evalua el 1ro si es true y con el false no hace nada */}
      <div className="card-grid">
        {
          images.map( img => (  //antes del custom hook era images.map, despues de la asignacion data:images puedo volver a usar images
            <GifGridItem 
              key={ img.id }
              {...img}
              // img={ img } 
            />
          ))
        }     

        {/* <ol>
          {
            images.map( img => (
              <li key={ img.id }> { img.title } </li>
            ))
          }
          ####################################
          aca tb puedo desestructurar y hacer 
            images.map( { id, title } => (
              <li key={ 
                id }> { title } </li>
            )) 
          ######################################  
        </ol> 
           <h3> { count } </h3>
        <button onClick={ ()=> setCount( count + 1 ) }></button>*/}
      </div> 
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
}

