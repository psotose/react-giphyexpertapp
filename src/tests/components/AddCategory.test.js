import react from 'react';
import '@testing-library/jest-dom'
import { shallow } from 'enzyme';
import { AddCategory } from "../../components/AddCategory";

describe('Testing AddCategory', () =>{

  const setCategories = jest.fn();
  let wrapper = shallow(< AddCategory setCategories={setCategories} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(< AddCategory setCategories={setCategories} />);
  });

  test('should be shown correctly', () =>{

    expect ( wrapper ).toMatchSnapshot();

  });

  test('should change the text box ', () =>{

    const input = wrapper.find('input');
    const value = "hola mundo";

    input.simulate('change', { target: { value } });
    expect( wrapper.find('p').text().trim() ).toBe( value );

  });

  test('should not post the info on submit', () => {

    wrapper.find('form').simulate('submit', { preventDefault(){} });
    expect( setCategories ).not.toHaveBeenCalled();
  });

  test('should call setCategories and clean out the text box', () => {
    const value = "hola mundo";
    wrapper.find('input').simulate('change', { target: { value } });
    wrapper.find('form').simulate('submit', { preventDefault(){} });
    expect ( setCategories ).toHaveBeenCalled();
    expect ( wrapper.find('input').prop('value') ).toBe('');
  });

});
