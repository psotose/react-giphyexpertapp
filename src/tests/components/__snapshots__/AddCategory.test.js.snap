// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing AddCategory should be shown correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <p>
     
     
  </p>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
